File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end(),[](int a, int b){
            string s1 = to_string(a);
            string s2 = to_string(b);
            return s1+s2 > s2+s1;
        });

        string res = "";
        for(int ele: nums){
            res += to_string(ele);
        }

        if(res[0] == '0')
            return "0";

        return res;
    }
};

File: 0179-largest-number/0179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {

        String[] str = new String[nums.length];
        for(int i = 0; i < nums.length; i++){
            str[i] = String.valueOf(nums[i]);
        }

        Arrays.sort(str, (a,b) -> {
            return (b+a).compareTo(a+b);
        });

        StringBuilder sb = new StringBuilder();
        for(String s: str)
            sb.append(s);
        
        if(sb.charAt(0) == '0')
            return "0";

        return sb.toString();

    }
}